modules/User.php

```diff
@@ -1,6 +1,108 @@
 <?php
 	class User {
-		function __construct() {
-			return;
+		public $email;
+		public function __construct($email) {
+			if (Utils::isValidEmail($email)) {
+				$this->email = $email;
+			} else {
+				return NULL;
+			}
+		}
+
+		private function getEncryptedPassword($password) {
+			if (isset($this->email)) {
+				return sha1(
+					substr($this->email, 0, strlen($this->email)) .
+					substr($password, 0, strlen($password)) .
+					substr($this->email, strlen($this->email)) .
+					substr($password, strlen($password))
+				);
+			}
+			return false;
+		}
+
+		public function getUserDetails($userData) {
+			return false;
+		}
+
+		public function isValidUser() {
+			global $DB_CONNECT;
+
+			$email = $this->email;
+
+			$stmt = $DB_CONNECT->prepare("SELECT `user_id` FROM `users` WHERE `email` = ?");
+			$stmt->bind_param('s', $email);
+			$stmt->execute();
+			$result = $stmt->get_result();
+			$row = $result->fetch_assoc();
+			$stmt->close();
+
+			if (!empty($row && isset($row['user_id']))) {
+				return $row['user_id'];
+			}
+			return false;
+		}
+
+		public function validateCredentials($password) {
+			global $DB_CONNECT;
+
+			$email = $this->email;
+
+			$stmt = $DB_CONNECT->prepare("SELECT * FROM `users` WHERE `email` = ?");
+			$stmt->bind_param('s', $email);
+			$stmt->execute();
+			$result = $stmt->get_result();
+			$row = $result->fetch_assoc();
+			$stmt->close();
+
+			$encryptedPassword = $this->getEncryptedPassword($password);
+			if (!empty($row) && $row != NULL && $row['email'] == $this->email && $row['password'] == $encryptedPassword) {
+				return true;
+			}
+			return false;
+		}
+
+		public function addUser($userData) {
+			global $DB_CONNECT;
+			if (
+				isset($userData['email']) &&
+				isset($userData['name']) &&
+				isset($userData['password']) &&
+				isset($userData['country']) &&
+				isset($this->email)
+			) {
+				if ($userData['email'] == $this->email) {
+					$username = strtolower(str_replace(' ', '_', $userData['name']));
+					$password = $this->getEncryptedPassword($userData['password']);
+					$stmt = $DB_CONNECT->prepare("
+						INSERT INTO `users` (
+							`email`,
+							`name`,
+							`password`,
+							`username`,
+							`country`
+						) VALUES (?, ?, ?, ?, ?)"
+					);
+					$stmt->bind_param (
+						'sssss',
```

Comment:
Here I am using prepared statement which is designed for sending safe query to db server, this can be done by escaping user input which is not part of the real query, and also checking the query. [This](https://stackoverflow.com/questions/18426172/what-does-bind-param-accomplishl) is what I have done.